modules/p2/ding_list/theme/theme.inc

```diff
@@ -166,52 +160,92 @@ function template_preprocess_ding_list_list(&$variables) {
+        $follow_list = ding_list_get_list(DING_LIST_TYPE_LISTS);
+        if ($element = $follow_list->hasElement($list->getId())) {
+          // If we have more than 1 remove form on a page Drupal will use the first
+          // occoring remove form everytime, unless they all got a unique ID, which
+          // is why the element ID is appended.
+          // See the ding_list_forms() hook in ding_list/include/form.inc.
+          $variables['content']['carousel_actions']['follow_form'] = drupal_get_form('ding_list_remove_element_form__' . $element->getId(), $element, t('Unfollow list'));
+        }
+        else {
+          $variables['content']['carousel_actions']['follow_form'] = drupal_get_form('ding_list_add_element_form', 'ding_list', $list->getId(), t('Add to lists I follow'));
         }
       }
+      break;
 
-      if (!empty($list->getNote())) {
-        $variables['content']['note'] = array(
-          '#prefix' => '<div class="ding-list-list__note">',
-          '#suffix' => '</div>',
-          '#markup' => $list->getNote(),
+    default:
+      if ($list->isOwner()) {
+        $variables['content']['elements'] = array(
+          '#theme' => 'table',
+          '#attributes' => array(
+            'id' => drupal_html_id('ding-list-list-elements'),
+            'class' => array(
+              'ding-list-list__elements', 
+              'ding-list-list__owner', 
+              'clearfix',
+            ),
+            'data-list-id' => $list->getId(),
+          ),
+          '#rows' => array(),
+          '#attached' => array(
+            'js' => array(
+              drupal_get_path('module', 'ding_list') . '/js/ding_list_tabledrag.js',
+            ),
+            'css' => array(
+              drupal_get_path('module', 'ding_list') . '/css/ding-list-tabledrag.css',
+            ),
+          ),
+        );
+        drupal_add_tabledrag(
+          $variables['content']['elements']['#attributes']['id'],
+          'order',
+          'sibling',
+          'ding-list-list__element-delta'
         );
       }
-
-      $variables['content']['elements'] = array(
-        '#theme' => 'table',
-        '#attributes' => array(
-          'id' => drupal_html_id('ding-list-list-elements'),
-          'class' => array('ding-list-list__elements'),
-          'data-list-id' => $list->getId(),
-        ),
-        '#rows' => array(),
-        '#attached' => array(
-          'js' => array(
-            drupal_get_path('module', 'ding_list') . '/js/ding_list_tabledrag.js',
-          ),
-          'css' => array(
-            drupal_get_path('module', 'ding_list') . '/css/ding-list-tabledrag.css',
+      else {
+        $variables['content']['elements'] = array(
+          '#theme' => 'table',
+          '#attributes' => array(
+            'id' => drupal_html_id('ding-list-list-elements'),
+            'class' => array(
+              'ding-list-list__elements', 
+              'ding-list-list__guest', 
```

Comment:
See comment regarding class naming above.

Reply:
Have added the ding-list-list__guest class, and used the same naming as the existing classes. Have changed the naming of ding-list-data--info, to ding-list-data__info to use same naming convention